util/llbutil/secretprovider/map_store.go

```diff
@@ -15,7 +17,15 @@ func NewMapStore(m map[string][]byte) secrets.SecretStore {
 
 // GetSecret gets a secret from the map store
 func (m mapStore) GetSecret(ctx context.Context, id string) ([]byte, error) {
-	v, ok := m[id]
+	q, err := url.ParseQuery(id)
+	if err != nil {
+		return nil, errors.New("failed to parse secret ID")
+	}
+
+	if q.Get("name") == "" {
+		return nil, errors.New("name parameter not found")
+	}
+	v, ok := m[q.Get("name")]
```

Comment:
should org or project be referenced here too?

How would a user deal with two Earthfiles referencing the same secret name; however one Earthfile defines a project, but the other one doesn't -- i.e. one is supplied locally, but the other one is cloud-based.

Reply:
update: after our chat I understand this is by design, and it's not possible to independently override one.